src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1984,7 +1994,16 @@ private void OnPreferencesWindowClick(object sender, RoutedEventArgs e)
             preferencesWindow = new PreferencesView(this);
             dynamoViewModel.OnPreferencesWindowChanged(preferencesWindow);
             preferencesWindow.WindowStartupLocation = WindowStartupLocation.CenterOwner;
-            preferencesWindow.ShowDialog();
+            preferencesWindow.Show();
+        }
+
+        public void EnableEnvironment(bool isEnabled)
+        {
+            this.titleBarGrid.IsEnabled = isEnabled;
```

Comment:
When the pref window is opened, these are already disabled, right? why do we need this method?

Reply:
yes, because it was being opened using ShowDialog(), but in this way I couldn't find how to detect when the user click outside the preferences, that's Why I'm using Show() instead so I need to disable/enable the main components of Dynamo explicitly

Reply:
so you mean `Show()` will set `PreferencesWindow.IsLoaded` to true?

Reply:
Using the ShowDialog() didn't allow me detect when the user click outside the preferences view (main goal), when we use it we don't worry about the state because it locks the Dynamo environment.

Reply:
@QilongTang with my last commit https://github.com/DynamoDS/Dynamo/pull/13973/commits/14c28d26e24ddfeb02a1aa807ba600ce82bac2d6 we won't worry about  enable/disable specific Dynamo parts if we add a new grid.